src/lib.rs

```diff
@@ -93,11 +95,18 @@ pub struct Infer {
 impl Infer {
     /// Initialize a new instance of the infer struct.
     pub fn new() -> Infer {
-        let mut v: Vec<(map::MatcherType, String, String, Matcher)> = Vec::new();
-
-        map::setup(&mut v);
+        Infer { mmap: Vec::new() }
+    }
 
-        Infer { mmap: v }
+    fn iter_matchers(&self) -> impl Iterator<Item = (&MatcherType, &str, &str, &Matcher)> {
+        let custom = self
+            .mmap
+            .iter()
+            .map(|(mt, mime, ext, matcher)| (mt, mime.as_str(), ext.as_str(), matcher));
+        MATCHER_MAP
+            .iter()
+            .map(|(mt, mime, ext, matcher)| (mt, *mime, *ext, matcher))
+            .chain(custom)
```

Comment:
Does it make more sense to chain `MATCHER_MAP` to `custom` iterator? Idea being that if clients add custom matchers, perhaps we should evaluate those first before we evaluate the standard ones. Just a thought.

Reply:
I fully agree but I chose to keep the current behavior (custom matchers were pushed to the vector). Since we all agree I will push a new commit to this PR that does the change, but you will need to bump the `Infer` release to a major number (0.2 -> 0.3) since this change might affect current users. I will document that in the API.

Reply:
We'll probably have to do a 0.3 release anyway if we add no_std support, but it's probably better to leave it as is for the moment just so we can do one last 0.2 release integrating those changes

Reply:
It is up to you guys, the commit is ready to be pushed to this PR or another one if you prefer.

Reply:
You are right... there is a semver-breaking implication to that change. Yup lets do the release dance and keep the breaking change out of this PR and release this change as a patch version; and then we can introduce the behaviour change in subsequent release where we're introducing API changes anyway.